src/Types/TypedTxOut.purs

```diff
@@ -64,99 +65,83 @@ import Cardano.Types.Value (Value)
 newtype TypedTxOutRef (a :: Type) (b :: Type) = TypedTxOutRef
   { txOutRef :: TransactionInput, typedTxOut :: TypedTxOut a b }
 
--- `DatumType a b` not needed but this replicates Plutus and provides extra
+-- `DatumType v d` not needed but this replicates Plutus and provides extra
 -- type safety.
-derive newtype instance (DatumType a b, Eq b) => Eq (TypedTxOutRef a b)
+derive newtype instance (DatumType v d, Eq d) => Eq (TypedTxOutRef v d)
 
 -- | Extract the `Address` of a `TypedTxOutRef`
 typedTxOutRefAddress
-  :: forall (a :: Type) (b :: Type)
-   . DatumType a b
-  => FromData b
-  => ToData b
-  => TypedTxOutRef a b
+  :: forall (v :: Type) (d :: Type)
+   . DatumType v d -- TODO: are there artificial constraints needed? (here and below)
```

Comment:
@klntsky number of "getters" for `TypedTxOutRef` and `TypedTxOut` types have these unneeded constraints `DatumType validator datum`. Are they needed or should I remove them?

Reply:
@Mr-Andersen 
Is there a reason to use these functions when the constraints are not satisfied?

Reply:
Values of type `TypedTxOut{,Ref}` correct only when this constraint holds. So, answering your question, code, using these getters with unsatisfied constraints, would be incorrect. I will experiment with adding smth like:

```
newtype TypedTxOut (validator :: Type) (datum :: Type) = DatumType validator datum => TypedTxOut
  { txOut :: TransactionOutput, data :: datum }
```

to see if alternatives possible